src/main/typed-ipc.ts

```diff
@@ -0,0 +1,125 @@
+import { BattleWithMetadata } from "@main/game/battle/battle-types";
+import { DownloadInfo } from "@main/content/downloads";
+import { EngineVersion } from "@main/content/engine/engine-version";
+import { GameVersion } from "@main/content/game/game-version";
+import { Info } from "@main/services/info.service";
+import { ipcRenderer as electronIpcRenderer, ipcMain as electronIpcMain, IpcMain, IpcMainEvent, IpcMainInvokeEvent, IpcRenderer, IpcRendererEvent, WebContents } from "electron";
+import { logLevels } from "@main/services/log.service";
+import { MapData } from "@main/content/maps/map-data";
+import { MultiplayerLaunchSettings } from "@main/game/game";
+import { NewsFeedData } from "@main/services/news.service";
+import { Replay } from "@main/content/replays/replay";
+import { Scenario } from "@main/content/game/scenario";
+import { Settings } from "@main/services/settings.service";
+import { TachyonEvent } from "tachyon-protocol";
+
```

Comment:
To start with I've just typed these as per the existing handle and listener functions whatever they happened to be.

Note: Some of these are returning un-necessary `| null | undefined`, one even returns `void[]` which is a little odd. We can tidy that up in the future.